### PR TITLE
Fix netbox pagination

### PR DESCRIPTION
**Pull request by wvandeun**
Fixes netbox pagination.
This fixes the same issue as #361. The owner of that PR didn't have the time to continue to work on the PR & I needed a solution so I implemented a solution myself.
I have discussed this @bdlamprecht and he was ok with the approach.

### CODE REVIEW

**Comment by dbarrosop** [comment]
Thanks for this work, can we create a new test without removing the old one? New code should make sure the old test also passes. Thanks again!

**Comment by wvandeun** [comment]
The old test still passes. 
I just deleted some lines which were commented out and not useful in any way. 
The signature of the get_inv function changed: I added a parameter to enable/disable pagination.

If you want I can just create another get_inv_pagination function, so that the old test is still exactly the same.

File: nornir/plugins/inventory/netbox.py [hunk 1]
```diff
@@ -34,19 +35,30 @@ def __init__(
...
-        r.raise_for_status()
-        nb_devices = r.json()
+        session = requests.Session()
+        session.headers.update({"Authorization": f"Token {nb_token}"})
+        session.verify = ssl_verify
+
+        # Fetch all devices from Netbox
+        # Since the api uses pagination we have to fetch until no next is provided
+
+        url = f"{nb_url}/api/dcim/devices/?limit=0"
```
Comment (dbarrosop) [inline nornir/plugins/inventory/netbox.py]: is `limit=0` needed?

**Comment by dbarrosop** [comment]
You are right, sorry, the diff mislead me and made me think you had deleted things you hadn't. It's all good then, just a minor question due to my lack of operational knowledge with netbox

File: nornir/plugins/inventory/netbox.py (hunk 1 above)
Comment (wvandeun) [reply nornir/plugins/inventory/netbox.py]: It is not strictly required. However if not specified then Netbox will return the number of records configured in PAGINATE_COUNT (50 by default). If limit is set to 0 then the Netbox will return the maximum number of records specified by MAX_PAGE_SIZE configuration.
50 is a bit low and would take quite some requests in a reasonable sized inventory.

Comment (wvandeun) [reply nornir/plugins/inventory/netbox.py]: https://netbox.readthedocs.io/en/latest/api/overview/#pagination

Comment (dbarrosop) [reply nornir/plugins/inventory/netbox.py]: ok, thanks for the clarification. Merging :)

**Comment by bdlamprecht** [comment]
Nice, great work @wvandeun!
All it takes is someone who knows what their doing and has the time.

**Pull request merged**